src/main/java/org/dcsa/ebl/service/impl/ShippingInstructionServiceImpl.java

```diff
@@ -291,6 +298,18 @@
             "number of originals is required for non electronic shipping instructions.");
       }
     }
+    if (Objects.nonNull(shippingInstructionTO.getIsElectronic())
+        && shippingInstructionTO.getIsElectronic()) {
+      if (Objects.isNull(shippingInstructionTO.getDocumentParties())
+          || shippingInstructionTO.getDocumentParties().isEmpty()) {
+        validationErrors.add("A documentParty with partyFunction=EBL is required for electronic shipping instructions.");
+      }
+      if (shippingInstructionTO.getDocumentParties().stream()
+          .noneMatch(x -> x.getPartyFunction().equals(PartyFunction.EBL))) {
+        validationErrors.add(
+            "For electronic Bill of Laden (isElectronic=true) - an EBL solution provider (PartyFunction=EBL) needs to be provided as a DocumentParty");
+      }
```

Comment:
Should we ensure that:

- There is exactly one EBL platform? (What would one do if the shipper listed two?)
- That the party also has an URL? (Or is that handled elsewhere?)

Reply:
Good point. You need to check the entire list. If multiple EBL platforms are listed - then send an error:
"Only 1 EBL solution provider can be specified in DocumentParties"

Reply:
Check has been changed to ensure that exactly one EBL is present in DocumentParty when `isElectronic` is true.